src/ucp/tag/rndv.c

```diff
@@ -457,15 +598,29 @@ static void ucp_rndv_handle_recv_contig(ucp_request_t *rndv_req, ucp_request_t *
     } else {
         if (rndv_rts_hdr->flags & UCP_RNDV_RTS_FLAG_PACKED_RKEY) {
             ucp_request_rndv_get_init(rndv_req);
-            UCS_PROFILE_CALL(uct_rkey_unpack, rndv_rts_hdr + 1,
-                             ucp_tag_rndv_rkey_bundle(rndv_req, 0));
+            ucp_tag_rndv_unpack_send_rkey(rndv_req, rndv_rts_hdr + 1);
+            ucs_assert_always(rndv_req->send.rndv_get.rkey != NULL);
+        }
+
+        if (!ucp_ep_is_stub(rndv_req->send.ep)) {
+            /* for regular, not stub, ep's we may resolve lanes here and
+             * set wakeup lane from resolved lanes */
+            ucp_tag_rndv_resolve_lanes(rndv_req);
+            rndv_req->send.lane = ucp_tag_rndv_get_lane_info(rndv_req, 0)->lane;
+        } else {
+            rndv_req->send.lane = UCP_NULL_LANE;
+        }
+
+        if (rndv_req->send.lane == UCP_NULL_LANE) {
```

Comment:
this is not needed now, right? Because `ucp_tag_rndv_resolve_lanes` should find the proper lane in either way (offload or not). BTW, if you remove this condition branch, make sure you don't assign `rndv_req->send.lane` to `UCP_NULL_LANE` in case of stub ep, otherwise you will get segfault while adding request to the pending queue on UCT ep.

Reply:
incorrect: in case if no one matched line is found - then pre-defined tag or rndv lane is used to ensure that progress works properly (to call progress callback). in this case rndv will fallback to AM mode

Reply:
in this case it is worth to use am lane regardless of `UCP_RNDV_RTS_FLAG_OFFLOAD` presence

Reply:
why? In this case we have just one lane capable of AM bcopy. When stub ep is switched to normal one, all pending operations are moved to that ep. You may check `ucp_ep_create_stub`

Reply:
yes, but stub EP is just one of the corner case, current implementation is the simplest and covers all possible cases (stub, no lanes found, tag offload falled back to rndv-get, etc)